snaps/reference/permissions.md

```diff
@@ -128,6 +128,40 @@ Specify this permission in the manifest file as follows:
 }
 ```
 
+### `endowment:name-lookup`
+
+:::flaskOnly
+:::
+
+To provide [custom name resolution](../features/custom-name-resolution.md), a Snap must request the
+`endowment:name-lookup` permission.
+This permission grants the Snap read-only access to user input or an address by exporting the
+[`onNameLookup`](../reference/entry-points.md#onnamelookup) entry point.
+
+This permission takes an object with two optional properties:
+
+- `chains` - an array of [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) chain IDs for which the Snap can resolve names and addresses. Pass this array to reduce overhead on your Snap by making sure it only receives requests for chains it can resolve.
+- `matchers` - an object that helps reduce overhead by further filtering the domains that are passed to your Snap. It must contain at least one of the following properties:
+
+  - `tlds` - an array of strings for top-level domains that the Snap supports.
+  - `schemes` - an array of strings for schemes that the Snap supports.
+
+Specify this permission in the manifest file as in the example below:
+
+```json title="snap.manifest.json"
+"initialPermissions": {
+  "endowment:name-lookup": {
+    "chains": ["eip155:1"],
+    "matchers": {
+      "tlds": ["crypto"],
+      "schemes": ["farcaster"]
+    }
+  }
+},
+```
+
+With the previous example, the Snap's [`onNameLookup` entry point](./entry-points.md#onnamelookup) would be called for domains such as `someuser.crypto` or `farcaster:someuser`, as long as the domain resolution is happening on Ethereum Mainnet.
```

Comment:
or **schemes such as** `farcaster:someuser`

Reply:
I think mainnet is always lowercase? Can we double-check that?

Reply:
<img width="1040" alt="image" src="https://github.com/MetaMask/metamask-docs/assets/3943143/46b4542e-937a-4c63-af11-79af6464c046">

on Ethereum.org it has a capital M. I'll fix scheme vs domain

Reply:
OK!

Reply:
We use capitalized "Mainnet" throughout the docs (and other Consensys docs). Should leave this for now for consistency, but open to making it lowercase in the future.